test/test_query.cpp

```diff
@@ -2384,7 +2384,7 @@ TEST(Query_TwoColumnsNumeric)
             size_t num_expected_matches = num_rows;
             if ((lhs_type == type_Mixed) != (rhs_type == type_Mixed)) {
                 // Only one prop is mixed
-                num_expected_matches = 6;
+                num_expected_matches = 5;
```

Comment:
I added a comment to this test. The number of matches changes in this test because I also changed `TestValueGenerator::convert_for_test<Mixed>` to return an ObjectId instead of an int in one case.